src/matter/common/MessageExchange.ts

```diff
@@ -15,9 +15,21 @@ import { Logger } from "../../log/Logger";
 import { NodeId } from "./NodeId";
 import { ByteArray } from "@project-chip/matter.js";
 import { SecureChannelProtocol } from "../session/secure/SecureChannelProtocol";
+import { MatterError } from "../../error/MatterError";
 
 const logger = Logger.get("MessageExchange");
 
+export class UnexpectedMessageResponseError extends MatterError {
+    public constructor(
+        public readonly message: string,
+        public readonly data: Message,
+    ) {
+        super();
```

Comment:
super(`(${MessageCodec.messageToString(data)}) ${message}`);
message is already a parameter of the constructor.